README.md

```diff
@@ -1,39 +1,431 @@
+[![Build Status](https://travis-ci.org/athityakumar/daru-io.svg?branch=master)](https://travis-ci.org/athityakumar/daru-io)
+[![Yard Docs](http://img.shields.io/badge/yard-docs-blue.svg)](http://www.rubydoc.info/github/athityakumar/daru-io/master/)
+[![Inline docs](http://inch-ci.org/github/athityakumar/daru-io.png)](http://inch-ci.org/github/athityakumar/daru-io)
+[![Code Climate](https://codeclimate.com/github/athityakumar/daru-io.png)](https://codeclimate.com/github/athityakumar/daru-io)
+[![Stories in Ready](https://badge.waffle.io/athityakumar/daru-io.png?label=ready&title=Ready)](https://waffle.io/athityakumar/daru-io?utm_source=badge)
+[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
-TODO: Delete this and the text above, and describe your gem
+A Ruby plugin-gem to [daru gem](https://github.com/SciRuby/daru), that extends support for many Import and
+Export methods of `Daru::DataFrame`. This gem is intended to help Data analyzing and Web Developer Rubyists,
+by serving as a general purpose conversion library that takes input in one format (say, JSON) and
+converts it another format (say, Avro) while making it incredibly easy to analyze the data.
 
-## Installation
+# Table of contents
 
-Add this line to your application's Gemfile:
+- [Installation](#installation)
+- *[Importers](#importers): [ActiveRecord](#activerecord-importer), [Avro](#avro-importer), [CSV](#csv-importer), [Excel](#excel-importer), [Excelx](#excelx-importer), [HTML](#html-importer), [JSON](#json-importer), [Mongo](#mongo-importer), [Plaintext](#plaintext-importer), [RData](#rdata-importer), [RDS](#rds-importer), [Redis](#redis-importer), [SQL](#sql-importer)*
+- *[Exporters](#exporters): [Avro](#avro-exporter), [CSV](#csv-exporter), [Excel](#excel-exporter), [JSON](#json-exporter), [RData](#rdata-exporter), [RDS](#rds-exporter), [SQL](#sql-exporter)*
+- [Contributing](#contributing)
+- [License](#license)
+
+# Installation
+
+[(Go to Table of Contents)](#table-of-contents)
+
+- If you're working with a Gemfile,
+
+    - Add this line to your application's Gemfile:
+
+    ```ruby
+    gem 'daru-io'
+    ```
+
+    - And then execute:
+
+    ```sh
+    bundle
+    ```
+
+- If you're NOT working with a Gemfile, simply install it yourself as:
+
+  ```sh
+  gem install daru-io
+  ```
+
+- Require `daru-io` gem in your application:
+
+```ruby
+require 'daru/io' #! Requires all Importers & Exporters
+require 'daru/io/importers' #! Requires all Importers and not Exporters
+require 'daru/io/importers/json' #! Requires only JSON Importer
+```
+
+*Note - Each IO module has it's own set of dependencies. Have a look at the [Importers](#importers) and [Exporters](#exporters) section for dependency-specific information.*
+
+# Importers
```

Comment:
Typically, it is only one first-level heading in Markdown ;)
And, also: maybe provide some generic info on importers? Like "When included, each importer provide `DataFrame.from_<format>` method, which is blah-blah"